Server/MirObjects/PlayerObject.cs

```diff
@@ -380,7 +396,16 @@ public void StopGame(byte reason)
             }
 
             Envir.Players.Remove(this);
-            CurrentMap.RemoveObject(this);
+
+            if (CurrentMap != null && CurrentMap.ValidPoint(CurrentLocation))
+            {
+                var cell = CurrentMap.GetCell(CurrentLocation);
+                if (cell?.Objects != null && cell.Objects.Contains(this))
```

Comment:
Recommend to revert this change back to how it was. None of these things should be possible at this stage, so its just more error catching.